src/enclave/rpc_sessions.h

```diff
@@ -418,10 +461,27 @@ namespace ccf
         if (udp)
         {
           LOG_DEBUG_FMT("New UDP endpoint at {}", id);
-          auto session = std::make_shared<QUICSessionImpl>(
-            rpc_map, id, listen_interface_id, writer_factory);
-          sessions.insert(std::make_pair(
-            id, std::make_pair(listen_interface_id, std::move(session))));
+          if (
+            per_listen_interface.app_protocol ==
+            ccf::ApplicationProtocol::CUSTOM)
+          {
+            // We know it's a custom protocol, but the session creation function
+            // hasn't been registered yet, so we keep a nullptr until the first
+            // udp::inbound message.
```

Comment:
The comment here is very helpful. What's the window where we might be accepting but haven't registered a session-creator function yet? Do we accept early in node startup before the app constructor is called, or are we wanting to permit even later protocol registration? If it's safe to do so, I'd prefer to return an error to this accept ("I can't accept this protocol yet") rather than storing a nullptr (and the associated checks that requires later), and quietly dropping all `recv` data. I understand that may be dependent on client behaviour though.

Reply:
The registration currently happens during app initialization (`init_handlers`), because I considered all of this application-specific. There's no problem with registering something earlier though, it's just less accessible to the application. I could add a log line, but I can't return a meaningful error to the client, because during this time we don't know what an error in the custom protocol would look like. For TCP, we should really just refuse the connection, but somehow we'd have to tell the host to refuse until it gets a signal from the application. For UDP there's no refusal/acceptance and replying to every random packet could open us up to amplification attacks, so I think just dropping those packets is the right thing to do.